Lib/timeSeries.py

```diff
@@ -58,16 +75,16 @@ def readUnrTxyz2(self, fileName):
 
                     self.name = line.split()[0]
 
-                decYear.append(line.split()[2])
-                x.append(line.split()[3])
-                y.append(line.split()[4])
-                z.append(line.split()[5])
-                sigX.append(line.split()[6])
-                sigY.append(line.split()[7])
-                sigZ.append(line.split()[8])
-                Cxy.append(line.split()[9])
-                Cyz.append(line.split()[10])
-                Cxz.append(line.split()[11])
+                decYear.append(float(line.split()[2]))
+                x.append(float(line.split()[3]))
+                y.append(float(line.split()[4]))
+                z.append(float(line.split()[5]))
+                sigX.append(float(line.split()[6]))
+                sigY.append(float(line.split()[7]))
+                sigZ.append(float(line.split()[8]))
+                Cxy.append(float(line.split()[9]))
+                Cyz.append(float(line.split()[10]))
+                Cxz.append(float(line.split()[11]))
```

Comment:
Two thoughts here:

1. You're calling `line.split()` every time you want to access a piece of it which is most-likely redoing the work of splitting the line. The files you're processing are large, and this could be a solid chunk of time. You could call `line.split()` once and store the values in a variable
2. This chunk of changes where you have to add `float(..)` to all your calls illustrates that you should probably have a function `process_line_value` that does this work for you. That way if you need to change the way you process all the values again, at another time, you can just change that one function.

Reply:
I agree with your suggestion in both 1 and 2.

This function is wicked fast so hadn't thought too much about ways to optimize but if I wrote a program to read in like 1000 of these would probably want it to be as optimized as possible.